#!/bin/sh -e
## 50_dedicated.dpatch by Peter S Galbraith <psg@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: made a true toggle; added autoload tag.

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad /home/psg/emacs/emacs-goodies-el/newfiles/emacs-goodies-el/elisp/emacs-goodies-el/dedicated.el emacs-goodies-el/elisp/emacs-goodies-el/dedicated.el
--- /home/psg/emacs/emacs-goodies-el/newfiles/emacs-goodies-el/elisp/emacs-goodies-el/dedicated.el	2003-11-12 20:51:41.000000000 -0500
+++ emacs-goodies-el/elisp/emacs-goodies-el/dedicated.el	2003-11-12 20:18:53.000000000 -0500
@@ -4,7 +4,7 @@
 
 ;; Author: Eric Crampton <eric@atdesk.com>
 ;; Maintainer: Eric Crampton <eric@atdesk.com>
-;; Version: 1.0.0
+;; Version: 1.1.0
 ;; Keywords: dedicated, buffer
 
 ;; This file is not part of GNU Emacs.
@@ -34,16 +34,30 @@
 ;;
 ;; Dedicated buffers will have "D" shown in the mode line.
 
+;;; History:
+;; 
+;; 2003-11-12 Peter S Galbraith <psg@debian.org>
+;;  V1.0.0 found on gnu.emacs.sources archives for 2000/04/12:
+;;   http://groups.google.com/groups?selm=izn1mzrs60.fsf%40elmo.atdesk.com
+;;  V1.1.0 made `dedicated-mode' a true toggle; added autoload tag and made
+;;   minor checkdoc edits.
+
 ;;; Code:
 
 (defvar dedicated-mode nil
-  "Mode variable for dedicated minor mode.")
+  "Mode variable for dedicated minor mode.
+Use the command `dedicated-mode' to toggle or set this variable.")
 (make-variable-buffer-local 'dedicated-mode)
 
+;;;###autoload
 (defun dedicated-mode (&optional arg)
-  "Dedicated minor mode."
+  "Toggle dedicated minor mode.
+With ARG, turn minor mode on if ARG is positive, off otherwise."
   (interactive "P")
-  (setq dedicated-mode (not dedicated-mode))
+  (setq hs-headline nil
+	dedicated-mode (if (null arg)
+                           (not dedicated-mode)
+                         (> (prefix-numeric-value arg) 0)))
   (set-window-dedicated-p (selected-window) dedicated-mode)
   (if (not (assq 'dedicated-mode minor-mode-alist))
       (setq minor-mode-alist
@@ -51,3 +65,5 @@
                   minor-mode-alist))))
 
 (provide 'dedicated)
+
+;;; dedicated.el ends here
